<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>946</width>
    <height>249</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="autoFillBackground">
   <bool>false</bool>
  </property>
  <property name="styleSheet">
   <string notr="true">background-color: rgb(255, 255, 255);</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QPushButton" name="keyButton">
    <property name="geometry">
     <rect>
      <x>60</x>
      <y>80</y>
      <width>61</width>
      <height>61</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgb(85, 170, 255);
border-color: rgb(0, 0, 0);</string>
    </property>
    <property name="text">
     <string>KEY</string>
    </property>
   </widget>
   <widget class="QPushButton" name="ShowComPortList">
    <property name="geometry">
     <rect>
      <x>400</x>
      <y>140</y>
      <width>201</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Update COM port lists</string>
    </property>
   </widget>
   <widget class="QPushButton" name="ConnectToKeyPort">
    <property name="geometry">
     <rect>
      <x>240</x>
      <y>110</y>
      <width>61</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Connect</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_8">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>0</y>
      <width>541</width>
      <height>61</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Bradley Hand ITC</family>
      <pointsize>23</pointsize>
      <weight>75</weight>
      <italic>false</italic>
      <bold>true</bold>
      <kerning>true</kerning>
     </font>
    </property>
    <property name="text">
     <string>SMÃ˜SBL Straight Key Client</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="keyPortInvert">
    <property name="geometry">
     <rect>
      <x>350</x>
      <y>140</y>
      <width>41</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Inv</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="sideTone">
    <property name="geometry">
     <rect>
      <x>620</x>
      <y>150</y>
      <width>131</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Enable Side Tone</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="toneFreqBox">
    <property name="geometry">
     <rect>
      <x>860</x>
      <y>150</y>
      <width>51</width>
      <height>22</height>
     </rect>
    </property>
    <property name="suffix">
     <string extracomment="Hz"/>
    </property>
    <property name="minimum">
     <number>500</number>
    </property>
    <property name="maximum">
     <number>900</number>
    </property>
    <property name="singleStep">
     <number>25</number>
    </property>
    <property name="value">
     <number>700</number>
    </property>
   </widget>
   <widget class="QLabel" name="label_9">
    <property name="geometry">
     <rect>
      <x>760</x>
      <y>150</y>
      <width>91</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Frequency (Hz)</string>
    </property>
   </widget>
   <widget class="QSlider" name="verticalSlider">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>80</y>
      <width>22</width>
      <height>101</height>
     </rect>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
    <property name="maximum">
     <number>100</number>
    </property>
    <property name="singleStep">
     <number>1</number>
    </property>
    <property name="value">
     <number>50</number>
    </property>
    <property name="orientation">
     <enum>Qt::Vertical</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label_10">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>190</y>
      <width>55</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Volume</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QComboBox" name="keyPortDevice">
    <property name="geometry">
     <rect>
      <x>400</x>
      <y>110</y>
      <width>201</width>
      <height>22</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>Select Key COM port</string>
     </property>
    </item>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>110</y>
      <width>91</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string>CW Key port</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="KeyPortName">
    <property name="geometry">
     <rect>
      <x>310</x>
      <y>110</y>
      <width>81</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>COM2</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="keyIP">
    <property name="geometry">
     <rect>
      <x>690</x>
      <y>110</y>
      <width>99</width>
      <height>22</height>
     </rect>
    </property>
    <property name="text">
     <string>127.0.0.1</string>
    </property>
   </widget>
   <widget class="QRadioButton" name="KeyOnCTS">
    <property name="geometry">
     <rect>
      <x>240</x>
      <y>140</y>
      <width>51</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>CTS</string>
    </property>
    <property name="checked">
     <bool>true</bool>
    </property>
    <attribute name="buttonGroup">
     <string notr="true">buttonGroup</string>
    </attribute>
   </widget>
   <widget class="QRadioButton" name="KeyOnDSR">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>140</y>
      <width>55</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>DSR</string>
    </property>
    <attribute name="buttonGroup">
     <string notr="true">buttonGroup</string>
    </attribute>
   </widget>
   <widget class="QSpinBox" name="keyNetPort">
    <property name="geometry">
     <rect>
      <x>800</x>
      <y>110</y>
      <width>85</width>
      <height>20</height>
     </rect>
    </property>
    <property name="maximum">
     <number>65535</number>
    </property>
    <property name="value">
     <number>6000</number>
    </property>
   </widget>
   <widget class="QPushButton" name="ConnectToKeyNetwork">
    <property name="geometry">
     <rect>
      <x>620</x>
      <y>110</y>
      <width>61</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Connect</string>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>130</x>
      <y>80</y>
      <width>481</width>
      <height>101</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgb(170, 255, 255);</string>
    </property>
    <property name="title">
     <string>COM Ports</string>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox_2">
    <property name="geometry">
     <rect>
      <x>610</x>
      <y>80</y>
      <width>301</width>
      <height>61</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgb(255, 235, 231);</string>
    </property>
    <property name="title">
     <string>Network connections</string>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox_3">
    <property name="geometry">
     <rect>
      <x>610</x>
      <y>140</y>
      <width>301</width>
      <height>41</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgb(226, 222, 255);</string>
    </property>
    <property name="title">
     <string/>
    </property>
   </widget>
   <widget class="QPushButton" name="SetKeyDelay">
    <property name="geometry">
     <rect>
      <x>131</x>
      <y>190</y>
      <width>93</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Auto key delay</string>
    </property>
   </widget>
   <widget class="QLCDNumber" name="packetLatencyMax">
    <property name="geometry">
     <rect>
      <x>600</x>
      <y>190</y>
      <width>51</width>
      <height>21</height>
     </rect>
    </property>
   </widget>
   <widget class="QLCDNumber" name="packetLatency">
    <property name="geometry">
     <rect>
      <x>380</x>
      <y>190</y>
      <width>51</width>
      <height>21</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>300</x>
      <y>190</y>
      <width>74</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Latency (ms)</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>190</y>
      <width>54</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Max (ms)</string>
    </property>
   </widget>
   <widget class="QLCDNumber" name="packetLatencyMin">
    <property name="geometry">
     <rect>
      <x>490</x>
      <y>190</y>
      <width>51</width>
      <height>21</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>440</x>
      <y>190</y>
      <width>51</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Min (ms)</string>
    </property>
   </widget>
   <widget class="QPushButton" name="toneButton">
    <property name="geometry">
     <rect>
      <x>60</x>
      <y>150</y>
      <width>61</width>
      <height>61</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgb(85, 170, 255);
border-color: rgb(0, 0, 0);</string>
    </property>
    <property name="text">
     <string>Zero
Beat
Tone</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="keyDelay">
    <property name="geometry">
     <rect>
      <x>230</x>
      <y>190</y>
      <width>61</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <number>50</number>
    </property>
    <property name="maximum">
     <number>1500</number>
    </property>
    <property name="singleStep">
     <number>10</number>
    </property>
    <property name="value">
     <number>300</number>
    </property>
   </widget>
   <zorder>SetKeyDelay</zorder>
   <zorder>packetLatencyMax</zorder>
   <zorder>packetLatency</zorder>
   <zorder>label_3</zorder>
   <zorder>label_5</zorder>
   <zorder>packetLatencyMin</zorder>
   <zorder>label_4</zorder>
   <zorder>groupBox_3</zorder>
   <zorder>groupBox_2</zorder>
   <zorder>groupBox</zorder>
   <zorder>keyButton</zorder>
   <zorder>ShowComPortList</zorder>
   <zorder>ConnectToKeyPort</zorder>
   <zorder>label_8</zorder>
   <zorder>keyPortInvert</zorder>
   <zorder>sideTone</zorder>
   <zorder>toneFreqBox</zorder>
   <zorder>label_9</zorder>
   <zorder>verticalSlider</zorder>
   <zorder>label_10</zorder>
   <zorder>keyPortDevice</zorder>
   <zorder>label_6</zorder>
   <zorder>KeyPortName</zorder>
   <zorder>keyIP</zorder>
   <zorder>KeyOnCTS</zorder>
   <zorder>KeyOnDSR</zorder>
   <zorder>keyNetPort</zorder>
   <zorder>ConnectToKeyNetwork</zorder>
   <zorder>toneButton</zorder>
   <zorder>keyDelay</zorder>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionSelect_COM_port">
   <property name="text">
    <string>Select COM port</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
 <buttongroups>
  <buttongroup name="buttonGroup"/>
 </buttongroups>
</ui>
